internal/gatewayapi/extensionserverpolicy.go

```diff
@@ -173,3 +188,11 @@ func (t *Translator) translateExtServerPolicyForGateway(
 	}
 	return found
 }
+
+// Appends status ancestors from newPolicy into aggregatedPolicy's list of ancestors.
+func MergeAncestorsForExtensionServerPolicies(aggregatedPolicy, newPolicy *unstructured.Unstructured) {
```

Comment:
Nit: This method is only used for test? Could we move this method to extensionserverpolicy_test.go and not export it?